plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetastoreClosure.java

```diff
@@ -87,13 +88,35 @@ public Set<HiveColumnStatisticType> getSupportedColumnStatistics(Type type)
 
     public PartitionStatistics getTableStatistics(String databaseName, String tableName)
     {
-        return delegate.getTableStatistics(getExistingTable(databaseName, tableName));
+        return getTableStatistics(databaseName, tableName, Optional.empty());
+    }
+
+    public PartitionStatistics getTableStatistics(String databaseName, String tableName, Optional<Set<String>> columns)
```

Comment:
Instead of `Optional<Set>` we could use just `Set` and `EmptySet`? Optional.empty and emptySet cases are treated the same.

Reply:
They are not treated the same. `Optional.empty()` means get stats for all table columns, empty set of columns means not column stats.